src/pykeen/models/base.py

```diff
@@ -1078,12 +1144,11 @@ def __init__(
             regularizer=regularizer,
             predict_with_sigmoid=predict_with_sigmoid,
         )
-        self.embedding_dim = embedding_dim
-        self.entity_embeddings = get_embedding(
-            num_embeddings=triples_factory.num_entities,
-            embedding_dim=self.embedding_dim,
-            device=self.device,
-        )
+        self.entity_embeddings = Embedding(
+            num=triples_factory.num_entities,
+            dim=self.embedding_dim,
+        ).to(device=self.device)
```

Comment:
Why not make it possible to give the device directly in the new class's init to better match the old interface?

Reply:
I actually experienced initialization to be significantly faster when always done on cpu and then moved to GPU.

Reply:
Really interesting insight!

Reply:
for a discussion about some of the reasons you might look into https://developer.nvidia.com/gpugems/gpugems3/part-vi-gpu-computing/chapter-37-efficient-random-number-generation-and-application